Core/include/Acts/Propagator/ConstrainedStep.hpp

```diff
@@ -154,6 +158,8 @@ class ConstrainedStep {
   std::array<double, 3> m_values = {kNotSet, kNotSet, kNotSet};
   /// the accuracy value - this can vary up and down given a good step estimator
   double m_accuracy = kNotSet;
+
+  constexpr double& value_(Type type) { return m_values[toUnderlying(type)]; }
```

Comment:
Nitpick: this is only used for setting (I think), so it could be a `setValue` function that takes the value as an argument instead of returning a reference.